pkg/build/controller/common/util_test.go

```diff
@@ -0,0 +1,156 @@
+package common
+
+import (
+	"fmt"
+	"reflect"
+	"sort"
+	"strings"
+	"testing"
+	"time"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	"k8s.io/kubernetes/pkg/runtime"
+	"k8s.io/kubernetes/pkg/types"
+
+	buildapi "github.com/openshift/origin/pkg/build/api"
+	buildclient "github.com/openshift/origin/pkg/build/client"
+	pruner "github.com/openshift/origin/pkg/build/prune"
+	buildutil "github.com/openshift/origin/pkg/build/util"
+	"github.com/openshift/origin/pkg/client/testclient"
+)
+
+func mockBuildConfig(name string) buildapi.BuildConfig {
+	appName := strings.Split(name, "-")
+	buildsToKeep := int32(2)
+	return buildapi.BuildConfig{
+		ObjectMeta: kapi.ObjectMeta{
+			Name:      fmt.Sprintf("%s-build", appName[0]),
+			Namespace: "namespace",
+			Labels: map[string]string{
+				"app": appName[0],
+			},
+		},
+		Spec: buildapi.BuildConfigSpec{
+			SuccessfulBuildsHistoryLimit: &buildsToKeep,
+			FailedBuildsHistoryLimit:     &buildsToKeep,
```

Comment:
use different numbers for the failed count and successful count.  by using the same number you have no way of knowing if your code is using the right field for the right build type.